app/Http/Controllers/ApiController.php

```diff
@@ -102,6 +104,69 @@ public function update(Request $request)
     }
     
     /**
+     * Updates the sensors of a device.
+     *
+     * @param  Request  $request
+     * @return Response
+     */
+    public function sensor(Request $request)
+    {
+        // Validate the request
+        $validator = Validator::make($request->all(), [
+            'uuid'          => 'required|string|max:255|exists:devices,uuid',
+            'token'         => 'required|string|max:60',
+            'sensor_data.*.name'   => 'required',
+            'sensor_data.*.type'   => 'required',
+            'sensor_data.*.value'  => 'required',
+        ]);
+        
+        // If validation fails, send the validation error back with status 400.
+        if ($validator->fails()) {
+            return response()->json(['data' => $validator->errors()], 400);
+        }
+        
+        // Get the device record.
+        $device = Device::getDeviceByUUID($request->input('uuid'));
+        
+        // If token doesnt match then send 401 unauthorized.
+        if ($request->input('token') != $device->token) {
+            return response()->json(['data' => 'Bad token.'], 401);
+        }
+        
```

Comment:
The device isn't a user and isn't authenticated with the auth library. We use the simple case for validation. Form request validation was added in Laravel 5 for complex validation scenarios. While not necessary for this case, I'll look into using the abstraction.

Reply:
The form validation has an authorize function which if it returns false will return a http error code and if true will continue as normal. It might use the auth library but it isn't necessary to check the token is valid. Simply placing your check inside there will work and instead of you manually returning a json response it will do it for you. I get what you are saying I just think it will be cleaner.

Reply:
I see what you are saying at the very least I think you should use the auto response with ->validate(). That way you don't need to manually check and return json upon an error.

Reply:
validate() is much cleaner and respects the Accepts: header from the client (returning JSON or HTML as needed). Addressed with commit dfcbce1